### PR TITLE
Mistral large2 llamaindex agentic rag

### PR DESCRIPTION
**Pull request by yhou-uk**
*Description of changes:*
Add example implementation of Agentic RAG using Mistral Large-2 model and Llamaindex

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb [hunk 1]
```diff
@@ -0,0 +1,1419 @@
...
+    "\n",
+    "In this notebook, we demonstrate an example of building an agentic RAG application using the LlamaIndex framework. This application serves as a technology discovery and research tool, using the Mistral Large 2 model via Bedrock Converse as the LLM to orchestrate agent flow and generate responses. It interacts with well-known websites, such as Arxiv, GitHub, and TechCrunch, and can access knowledge bases containing documentation and internal knowledge.\n",
+    "\n",
+    "This application can be further expanded to accommodate broader use cases requiring dynamic interaction with internal and external APIs, as well as the integration of internal knowledge bases to provide more context-aware responses to user queries.\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "- Until October 31, the Mistral Large 2 model is only available in the `us-west-2` region.\n",
```
Comment (tugglep) [inline notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb:22]: I would change 'until...' to something like as of the date of this writing... or something. Until makes it sound like on Oct 31st something is changing.

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb [hunk 2]
```diff
@@ -0,0 +1,1419 @@
...
+    "- Create a SageMaker notebook instance and select `ml.t3.medium` as the instance type.\n",
+    "- Create a new SageMaker execution role and grant it Bedrock full access.\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Architecture\n",
+    "\n",
+    "This solution uses the LlamaIndex framework to build an agent flow with two main components: AgentRunner and AgentWorker. The AgentRunner serves as an orchestrator that manages conversation history, creates and maintains tasks, executes task steps, and provides a user-friendly interface for interactions. The AgentWorker handles the step-by-step reasoning and task execution.\n",
```
Comment (tugglep) [inline notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb:30]: can we link to the LlamaIndex documentation for agentrunner and agentworker?

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb [hunk 3]
```diff
@@ -0,0 +1,1419 @@
...
+    {
+     "name": "stdin",
+     "output_type": "stream",
+     "text": [
+      "User:  exit\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Simple chatbot UI. Entre \"exit\" to quit. \n",
```
Comment (tugglep) [inline notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb:480]: typo - enter

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb [hunk 4]
```diff
@@ -0,0 +1,1419 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ad478b4b-350c-47b5-a31f-2ebbbc864935",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "This notebook shows how we can combine LLMs (Mistral Large 2), internet searching tools, and knowledge bases to build an intelligent research helper. We can see how this solution works well for finding and understanding technical information, and it can easily be made more powerful by adding more data sources and features."
```
Comment (tugglep) [inline notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb:788]: make sure all outputs are cleared

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb (hunk 1 above)
Comment (yhou-uk) [reply notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb]: Addressed in the new commit.

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb (hunk 3 above)
Comment (yhou-uk) [reply notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb]: Addressed in the new commit.

File: notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb (hunk 4 above)
Comment (yhou-uk) [reply notebooks/Mistral_Large2_agentic_rag_with_llamaindex.ipynb:603]: Addressed in the new commit.

**Pull request merged**